test/e2e/upgrade_test.go

```diff
@@ -203,6 +204,14 @@ var _ = Describe("BMO Upgrade", func() {
 		framework.TryAddDefaultSchemes(scheme)
 		metal3api.AddToScheme(scheme)
 		upgradeClusterProxy = framework.NewClusterProxy("bmo-e2e-upgrade", kubeconfigPath, scheme)
+
+		bmoIronicNamespace, bmoIronicCancelWatches = framework.CreateNamespaceAndWatchEvents(ctx, framework.CreateNamespaceAndWatchEventsInput{
```

Comment:
It will be strange if we create the namespace if the user asked to not deploy BMO/Ironic. Why do we need to create it? Is it to watch the events?

Reply:
Ok so we don't really have to create the namespace, just clean it up, right? I think it would be better to not create it. We can still delete it after the test.
If we create it like this we will get an error if the user already created it (using existing cluster and already deployed Ironic).

Reply:
What I was aiming to is to be able to delete it before the test as well, since the cleanup might fail after the previous test. But yes, might be creating it separately is not necessary, I simply did that to be able to use your cleanup() function.